client/src/redux/actionCreators/location.ts

```diff
@@ -0,0 +1,57 @@
+import axios from 'axios';
+import { Dispatch } from 'redux';
+
+import { ActionType, Action } from '../actionTypes/location';
+
+export const searchLocations = (query: string) => async (dispatch: Dispatch<Action>) => {
+  dispatch({ type: ActionType.SEARCH_LOCATION_PENDING });
+
+  try {
+    const data = await axios.get(`http://localhost:3000/forecaApi/location/search/${query}`, {
+      withCredentials: true
+    });
+
+    dispatch({ type: ActionType.SEARCH_LOCATION_SUCCESS, payload: data.data.locations });
+  } catch (err) {
+    if (err instanceof Error) {
+      dispatch({ type: ActionType.SEARCH_LOCATION_FAIL, payload: err.message });
+    }
+  }
+};
+
+export const getLocationInfo = (query: string) => async (dispatch: Dispatch<Action>) => {
+  dispatch({ type: ActionType.GET_LOCATION_INFO_PENDING });
+
+  try {
+    const data = await axios.get(`http://localhost:3000/forecaApi/location/${query}`, {
+      withCredentials: true
+    });
+
+    dispatch({ type: ActionType.GET_LOCATION_INFO_SUCCESS, payload: data.data });
+  } catch (err) {
+    if (err instanceof Error) {
+      dispatch({ type: ActionType.GET_LOCATION_INFO_FAIL, payload: err.message });
+    }
+  }
+};
+
+export const getLocationCurrWeather = (query: string) => async (dispatch: Dispatch<Action>) => {
+  dispatch({ type: ActionType.GET_LOCATION_CURRENT_WEATHER_PENDING });
+
+  try {
+    const data = await axios.get(
+      `http://localhost:3000/forecaApi/current/${query}&tempunit=C&windunit=MS`,
```

Comment:
Why you don't use base url in const? 
Example: const BASE_USE = 'http://localhost:3000/forecaApi/'
I think it will be better

Reply:
I will, but later. That whole function needs refactoring, because all these actionCreators have similar and/or duplicate code.

Reply:
agree with it, please move url to separate variable